rest/route/github.go

```diff
@@ -222,23 +224,58 @@ func (gh *githubHookApi) Run(ctx context.Context) gimlet.Responder {
 			}
 		}
 
-	// This case merely logs. EVG-19964 will add logic to create a version from the GitHub merge queue.
 	case *github.MergeGroupEvent:
-		grip.Debug(message.Fields{
+		var msg string
+		if disableMergeGroup {
+			msg = "merge group received, skipping"
+		} else {
+			msg = "merge group received, attempting to queue"
+		}
+		grip.Info(message.Fields{
 			"source":   "GitHub hook",
 			"msg_id":   gh.msgID,
 			"event":    gh.eventType,
 			"org":      event.GetOrg(),
 			"repo":     event.GetRepo(),
 			"base_sha": event.GetMergeGroup().GetBaseSHA(),
 			"head_sha": event.GetMergeGroup().GetHeadSHA(),
+			"message":  msg,
 		})
 
+		// This will be removed when EVG-19964 is ready.
+		if disableMergeGroup {
+			return gimlet.NewJSONResponse(struct{}{})
+		}
+		if err := gh.AddIntentForGithubMerge(event.GetMergeGroup()); err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"source":   "GitHub hook",
+				"msg_id":   gh.msgID,
+				"event":    gh.eventType,
+				"org":      event.GetOrg(),
+				"repo":     event.GetRepo(),
+				"base_sha": event.GetMergeGroup().GetBaseSHA(),
+				"head_sha": event.GetMergeGroup().GetHeadSHA(),
+				"message":  "can't add intent",
+			}))
+			return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "adding patch intent"))
+		}
```

Comment:
This will never run. What is the intent behind committing it?  Is it mostly for documentation?

Reply:
It'll run when we remove the feature flag. The idea is not to expose this code path until we're ready.